tests/ssh/base.py

```diff
@@ -1,40 +1,50 @@
 import unittest
+import mock
 
 from netengine.backends.ssh import SSH
 from netengine.exceptions import NetEngineError
 
 from ..settings import settings
+from ..static import MockOutputMixin
 
 
 __all__ = ['TestSSH']
 
 
-class TestSSH(unittest.TestCase):
-
-    def setUp(self):
-        self.host = settings['base-ssh']['host']
-        self.username = settings['base-ssh']['username']
-        self.password = settings['base-ssh'].get('password', '')
-        self.port = settings['base-ssh'].get('port', 22)
-        
+class TestSSH(unittest.TestCase, MockOutputMixin):
+    @mock.patch('paramiko.SSHClient.connect')
+    def setUp(self, mocked_connect):
+        self.host = 'test-host.com'
+        self.username = 'test-user'
+        self.password = 'test-password'
+        self.port = 22
         self.device = SSH(self.host, self.username, self.password, self.port)
         self.assertTrue(self.device.__netengine__)
         self.device.connect()
-    
-    def test_validate_negative_result(self):
```

Comment:
Is this test required anymore?

Reply:
Could you please share some context about why would it not be required?
I mean we should want to test negative results too, correct?

Reply:
This test checks if we can log into a device with wrong credentials. Since we will not be using a real device anymore there is no reason to check this here. We should assume that the SSH library we are using already does this, what do you think? Does my explanation make sense? :smile:

Reply:
Hmm, although, it makes sense that the dependency should be checking this...
but I'd defer it to someone else for final word on this! :smile:

Reply:
I think the best thing to do in this case is to mock the underlying library call to raise the exception which is typicall raised when wrong credentials are passed and ensure our library behaves as we defined

Reply:
Done, added back (https://github.com/openwisp/netengine/pull/48/commits/a87cc398b37aeda9bb619d3f145c0795bc2221e8)